app/models/ems_cluster.rb

```diff
@@ -328,9 +328,14 @@ def register_host(host)
     end
   end
 
-  def self.node_types
-    return :non_openstack unless openstack_clusters_exists?
-    non_openstack_clusters_exists? ? :mixed_clusters : :openstack
+  cache_with_timeout(:node_types) do
+    if !openstack_clusters_exists?
+      :non_openstack
+    elsif non_openstack_clusters_exists?
+      :mixed_clusters
+    else
+      :openstack
+    end
```

Comment:
I'm ok with this but it still feels like this is wrong.  We should have cross cloud terminology, a combination of all the things, or a place for them to choose the wording.  The fact that we query the database, have lots of backend and UI code to try to figure out what to call things in the UI amazes me. cc @chessbyte